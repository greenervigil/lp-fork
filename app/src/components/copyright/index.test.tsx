import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';

import Copyright from '.';
import { faker } from '@faker-js/faker';

describe('Copyright', () => {
    describe('component', () => {
        it('should render', () => {
            const { baseElement } = render(<Copyright year="2000" />);
            expect(baseElement).toMatchSnapshot();
            expect(screen.getByTestId('copyright')).toBeInTheDocument();
        });

        it('should render the year', () => {
            const date = faker.date.future();
            const year = date.getFullYear();

            render(<Copyright year={year} />);
            expect(screen.getByTestId('copyright')).toHaveTextContent(
                year.toString()
            );
        });
    });
});
