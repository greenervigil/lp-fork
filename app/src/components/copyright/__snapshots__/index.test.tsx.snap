// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copyright component should render 1`] = `
<body>
  <div>
    <p
      class="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
      data-testid="copyright"
    >
      Â© 
      2000
       Learning Portrait
    </p>
  </div>
</body>
`;
